src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1901,9 +1906,8 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $container->registerAliasForArgument($name, HttpClientInterface::class);
 
             if ($hasPsr18) {
-                $container->register('psr18.'.$name, Psr18Client::class)
-                    ->setAutowired(true)
-                    ->setArguments([new Reference($name)]);
+                $container->setDefinition('psr18.'.$name, new ChildDefinition('psr18.http_client'))
+                    ->replaceArgument(0, new Reference($name));
```

Comment:
this change is not related to httplug - it's just better way to wire the psr18 named aliases